### PR TITLE
Prevent the deletion of already redeemed offline purchases via form [PD-911]

### PR DESCRIPTION
**Pull request by almathew**
All tests pass

### CODE REVIEW

File: postajob/views.py [hunk 1]
```diff
@@ -767,6 +767,11 @@ class OfflinePurchaseFormView(PostajobModelFormMixin, RequestFormViewBase):
     update_name = 'offlinepurchase_update'
     delete_name = 'offlinepurchase_delete'
 
+    def delete(self):
```
Comment (tdphillips) [inline postajob/views.py]: Does this need to be decorated as well?

File: postajob/tests/test_views.py [hunk 2]
```diff
@@ -300,6 +300,35 @@ def test_purchasedjob_delete(self):
...
+        data = dict(self.purchasedjob_form_data)
+        data['is_expired'] = False
+
+        response = self.client.post(reverse('purchasedjob_update',
+                                            kwargs=kwargs),
+                                    data=self.purchasedjob_form_data,
+                                    follow=True)
+        self.assertEqual(response.status_code, 200)
+
+        # Confirm that the job remains expired despite the attempt to
```
Comment (aspidites) [inline postajob/tests/test_views.py]: So are we not expecting a user to ever check the expired box and uncheck it later?

Comment (almathew) [inline postajob/tests/test_views.py]: This only happens if the job is past the max_expiration_date. Hence the changes to job on 313-316

Comment (aspidites) [inline postajob/tests/test_views.py]: Ahhh, I misread the comment. You're enforcing the checkbox irrespective of the actual date. Derp. :+1:

**Pull request merged**